app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,65 @@
+class TasksController < ApplicationController
+  before_action :logged_in_supervisor
+  before_action :load_course_subject, :load_subject, except: :destroy
+  before_action :load_task, except: %i(new create)
+
+  def new
+    @task = @subject.tasks.build
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  def create
+    @task = @subject.tasks.build task_params
+    respond_to do |format|
+      @ers = @task.errors.full_messages unless @task.save
+      format.js
+    end
+  end
+
+  def edit
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @task.update_attributes task_params
+        @new_task = Task.find_by id: @task.id
```

Comment:
new_task bị nil thì sao cậu ơi

Reply:
Mấy cái liên quan đến task này em dùng ajax hết á chị.
Đoạn này sau khi update thì em muốn lấy object đã có sự thay đổi để replace nó lên view hiện luôn chứ không reload hay render hết toàn bộ colection tại nên mới phải find_by thêm 1 lần. Trươcs đó em đã có gọi call_back load_task cho hàm update này rồi nên khi vào đc hàm update này có nghĩa là task nó đã luôn tồn tại rồi nên em không có check nil nữa.